localstack/services/ssm/ssm_listener.py

```diff
@@ -64,6 +71,43 @@ def get_params_and_secrets(names):
     return result
 
 
+def get_params_by_path_with_labels(path='', param_filters=None, labels_to_filter=None, recursive=False):
+    ssm_client = aws_stack.connect_to_service('ssm')
+    result = {'Parameters': []}
+    filters = [{'Key': 'Path', 'Values': [path]}]
+    filters.extend(param_filters)
+    if recursive:
+        filters[0]['Option'] = 'Recursive'
+
+    def filter_by_label(param=None, labels=None):
+        for label in param['Labels']:
+            if label in labels:
+                return param
+    try:
+        # Get all the params in the path
+        params_in_path = ssm_client.describe_parameters(ParameterFilters=filters)['Parameters']
+
+        # Get parameter with all its labels (for all the parameters in params_in_path)
+        # Labels of the parameters can be obtained by calling get_parameter_history with parameter name
+        all_params = []
+        for params in params_in_path:
+            all_params.extend(ssm_client.get_parameter_history(Name=params['Name'])['Parameters'])
+
+        # Filter the params with matched labels
+        filtered_params = list(filter(lambda param: filter_by_label(param=param, labels=labels_to_filter), all_params))
+
+        # Get details of the filtered params to return
+        # This step is needed because get_parameter_history doesn't return parameter's ARN
+        details_of_filtered_params = list(map(lambda param: ssm_client.get_parameter(Name=param['Name'])['Parameter'],
+            filtered_params))
+        result['Parameters'].extend(details_of_filtered_params)
+    except Exception as e:
+        LOG.info('Unable to get ssm parameters by path and filter by labels : %s %s' % (e,
```

Comment:
nit: Let's better use upper-case for service acronyms: "ssm parameters" -> "SSM parameters"